airbyte-webapp/src/components/connection/CatalogTree/next/StreamDetailsPanel/StreamFieldsTable/SyncFieldCell/SyncFieldCell.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { CheckBox } from "components/ui/CheckBox";
+import { Tooltip } from "components/ui/Tooltip";
+
+import { SyncSchemaField, SyncSchemaFieldObject } from "core/domain/catalog";
+import { SyncMode, DestinationSyncMode } from "core/request/AirbyteClient";
+
+interface SyncFieldCellProps {
+  field: SyncSchemaField;
+  checkIsCursor: (path: string[]) => boolean;
+  checkIsPrimaryKey: (path: string[]) => boolean;
+  isFieldSelected: boolean;
+  handleFieldToggle: (fieldPath: string[], isSelected: boolean) => void;
+  syncMode?: SyncMode;
+  destinationSyncMode?: DestinationSyncMode;
+}
+
+export const SyncFieldCell: React.FC<SyncFieldCellProps> = ({
+  checkIsCursor,
+  checkIsPrimaryKey,
+  isFieldSelected,
+  field,
+  handleFieldToggle,
+  syncMode,
+  destinationSyncMode,
+}) => {
+  const isNestedField = SyncSchemaFieldObject.isNestedField(field);
+  const isCursor = checkIsCursor(field.path);
+  const isPrimaryKey = checkIsPrimaryKey(field.path);
+  const isDisabled =
+    (syncMode === SyncMode.incremental && isCursor) ||
+    (destinationSyncMode === DestinationSyncMode.append_dedup && isPrimaryKey) ||
+    isNestedField;
+
+  const renderDisabledReasonMessage = useCallback(() => {
+    if (isNestedField) {
+      return <FormattedMessage id="form.field.sync.nestedFieldTooltip" values={{ fieldName: field.path[0] }} />;
+    }
+    if (isPrimaryKey) {
+      return <FormattedMessage id="form.field.sync.primaryKeyTooltip" />;
+    }
+    if (isCursor) {
+      return <FormattedMessage id="form.field.sync.cursorFieldTooltip" />;
+    }
+    return null;
+  }, [isCursor, isPrimaryKey, isNestedField, field.path]);
+
+  return (
+    <>
+      {!isDisabled && (
+        <CheckBox
+          checkboxSize="sm"
+          checked={isFieldSelected}
+          onChange={() => handleFieldToggle(field.path, !isFieldSelected)}
+        />
+      )}
+      {isDisabled && (
+        <Tooltip control={<CheckBox checkboxSize="sm" disabled checked={isFieldSelected} readOnly />}>
```

Comment:
I think I understand, and I think I generally agree when there's a lot of JSX, but I'm not sure if it would be any clearer in this instance. This component's render function basically does one thing only: checks the boolean `isDisabled` and returns one or the other bit of JSX. If I split that into another variable, wouldn't that same logic be repeated in the variable? What would the benefit be?